### PR TITLE
Support signed declare transaction

### LINKED ISSUE #145: Support signing and preparing declare transactions through Account
Currently, `Declare` transactions have to be manually prepared and signed. This flow should be improved to allow easy Declare preparation and signing through `Account` interface.

### PR DESCRIPTION
**Pull request by cptartur**
## Describe your changes

This PR adds support for signing declare transactions and sending them.

## Linked issues

Closes #145 

## Breaking changes

- [ ] This issue contains breaking changes

### CODE REVIEW

**Comment by cptartur** [comment]
Thing to consider: with current supported version of the RPC spec, it's impossible to send signed declare transactions. Starknet-jvm will first have to be updated to support RPC 0.2.0 for these to be able to work.

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Transaction.kt [hunk 1]
```diff
@@ -144,7 +143,24 @@ data class DeclareTransaction(
     override val nonce: Felt,
 
     override val type: TransactionType = TransactionType.DECLARE,
-) : Transaction()
+
+    private val contractDefinition: ContractDefinition? = null,
+) : Transaction() {
+    @Throws(ConvertingToPayloadFailedException::class)
```
Comment (Solpatium) [inline lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Transaction.kt:149]: I think we have toPayload in other places as well. I think we can resuse it there as well?

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/TransactionPayload.kt [hunk 2]
```diff
@@ -31,8 +32,17 @@ data class DeployTransactionPayload(
...
 data class DeclareTransactionPayload(
     val contractDefinition: ContractDefinition,
+    val senderAddress: Felt,
     val maxFee: Felt,
     val nonce: Felt,
     val signature: Signature,
     val version: Felt,
-)
+) {
+    constructor(
```
Comment (Solpatium) [inline lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/TransactionPayload.kt:41]: Why is this needed?

**Review by Solpatium — APPROVED** [review]

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/TransactionPayload.kt (hunk 2 above)
Comment (cptartur) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/TransactionPayload.kt:41]: I guess it's only used in tests using old declare transactions (unsigned). I'll remove this as we don't really want to support tx version 0.

File: lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Transaction.kt (hunk 1 above)
Comment (cptartur) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Transaction.kt:149]: I'll try to think of a way of making `toPayload` more defined and reusable.

Comment (cptartur) [reply lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/Transaction.kt:155]: I don't think this comment still applies?

**Comment by cptartur** [comment]
Note that the Rpc provider doesn't support sending signed Declare txs. I've added an error for that so it doesn't fail silently/is ignored. This will have to be addressed once rpc 0.2.1 is supported, shouldn't require any changes to the Account itself however.

File: lib/src/test/kotlin/starknet/account/StandardAccountTest.kt [hunk 3]
```diff
@@ -145,6 +143,34 @@ class StandardAccountTest {
...
+            classHash,
+            ExecutionParams(nonce, Felt(1000000000000000)),
+        )
+        val request = provider.declareContract(declareTransactionPayload)
+        val result = request.send()
+        val receipt = provider.getTransactionReceipt(result.transactionHash).send()
+
+        assertNotNull(result)
+        assertNotNull(receipt)
+        println(receipt)
```
Comment (THenry14) [inline lib/src/test/kotlin/starknet/account/StandardAccountTest.kt:170]: 👀

**Review by THenry14 — APPROVED** [review]
lgtm!

**Review by Solpatium — APPROVED** [review]
